source/initial-dns-seedlist-discovery/tests/sharded/srvMaxHosts-equal_to_srv_records.yml

```diff
@@ -0,0 +1,13 @@
+# When srvMaxHosts equals the number of SRV records, all hosts are added to the
+# seed list.
+uri: "mongodb+srv://test1.test.build.10gen.cc/?srvMaxHosts=2"
+numSeeds: 2
+seeds:
+    - localhost.test.build.10gen.cc:27017
+    - localhost.test.build.10gen.cc:27018
+hosts:
+    - localhost:27017
+    - localhost:27018
```

Comment:
@benjirewis: Since you mentioned these assertions fail in the Go driver

I see that [tests/load-balanced/loadBalanced-true-txt.yml](https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/tests/load-balanced/loadBalanced-true-txt.yml#L5) includes the following note:

> In LB mode, the driver does not do server discovery, so the hostname does not get resolved to localhost:27017.

When I wrote these tests, I was assuming that the mongos host name would be corrected to the canonical host reported in the `hello` response as part of SDAM, as is done in the replica set tests (e.g. [tests/replica-set/loadBalanced-false-txt.yml](https://github.com/mongodb/specifications/blob/master/source/initial-dns-seedlist-discovery/tests/replica-set/loadBalanced-false-txt.yml)).

That said, we didn't have existing tests for sharded clusters so I very well could have made a mistake here. I'll note that the [Polling SRV tests](https://github.com/mongodb/specifications/blob/master/source/polling-srv-records-for-mongos-discovery/tests/README.rst) discuss changing DNS records and go on to say "verify that the new list of hosts is present". They don't actually specify what the hosts should be, so we might assume that those host names should be identical to the seeds provided by DNS.

AFAIK, the only mention of host name validation in the SDAM spec talks about it from the context of replica set connections:

 * [Clients use the hostnames listed in the replica set config, not the seed list](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#clients-use-the-hostnames-listed-in-the-replica-set-config-not-the-seed-list)
 * [Using me field to detect seed list members that do not match host names in the replica set configuration](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#using-me-field-to-detect-seed-list-members-that-do-not-match-host-names-in-the-replica-set-configuration)

When [TopologyType explanations](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#topologytype-explanations) discussed the sharded topology, it only talks about keeping servers with the type Unknown or Mongos and removing all other servers. There is no mention of host name validation like we find in **updateRSWithoutPrimary**:

```
if description.address != description.me:
    remove this server from topologyDescription and stop monitoring it
    return
```

Given that, I think the SDAM spec's silence on the matter for mongos suggests that host name canonicalization does _not_ take place like it does for replica sets. Perhaps @ShaneHarvey or @nbbeeken can chime in and confirm.

Reply:
Sounds good, thanks for investigating. The Go driver implementation (still in review), does currently [use a modified version of these sharded tests](https://github.com/mongodb/mongo-go-driver/pull/764/files#diff-a5930af71de27d03199a7a6d483d4098cc02aba763d4a68b5f1d5659e86d60bdR8-R10) with the non-canonicalized host names.

Reply:
Updated all sharded tests to use the same host names in `seeds` and `hosts`.